### PR TITLE
Updated Go-based Operators documentation

### PR DESCRIPTION
**Pull request by sheriff-rh**
Docs JIRA:
https://issues.redhat.com/browse/OSDOCS-1090

OSDK JIRA: 
https://issues.redhat.com/browse/OSDK-865

Applies to 4.4+.

### CODE REVIEW

File: modules/osdk-cli-reference-generate.adoc [hunk 1]
```diff
@@ -8,6 +8,33 @@
...
+`deploy/crds/__crd.yaml`. OpenAPI V3 validation YAML is generated as a
+`validation` object.
+
+.`generate crds` flags
+[options="header",cols="1,3"]
+|===
+|Flag |Description
+
+|`--crd-version` (string)
+|CRD version to generate (default "v1")
```
Comment (adellape) [inline modules/osdk-cli-reference-generate.adoc:24]:
```
`v1`
```

File: modules/osdk-cli-reference-generate.adoc [hunk 2]
```diff
@@ -8,6 +8,33 @@
...
+|`-h`, `--help`
+|Help for `generate crds`
+|===
+
+.Example output
+----
+$ operator-sdk generate crds
+$ tree deploy/crds
+├── deploy/crds/app.example.com_v1alpha1_appservice_cr.yaml
+├── deploy/crds/app.example.com_appservices_crd.yaml
```
Comment (adellape) [inline modules/osdk-cli-reference-generate.adoc:35]:
```
└──
```

**Comment by sheriff-rh** [comment]
@jianzhangbjz PTAL, thanks!

**Comment by jianzhangbjz** [comment]
@joelanford Hi, what's the version of the operator-sdk for the OCP 4.5? I find the latest operator-sdk has removed the `Go` based operator.

**Comment by jianzhangbjz** [comment]
Or this 4.5 release doc based on the operator-sdk [v0.8.1](https://github.com/operator-framework/operator-sdk/tree/v0.8.1) according to [ocp-release-operator-sdk/release-4.5](https://github.com/openshift/ocp-release-operator-sdk/tree/release-4.5#quick-start)?

**Comment by jianzhangbjz** [comment]
@joelanford  I test this doc with the v0.17.1 version but got the below errors. Is it a bug? Thanks!
```console
[root@preserve-olm-env foo-operator]# operator-sdk generate crds
INFO[0000] Running CRD generator.                       
FATA[0000] error generating CRDs from APIs in pkg/apis: error generating CRD manifests: error reading CRD cache dir deploy/crds: open deploy/crds: file does not exist

[root@preserve-olm-env foo-operator]# ls deploy/crds/
foo.example.com_foos_crd.yaml  foo.example.com_v1alpha1_foo_cr.yaml
```
Versions:
```console
[root@preserve-olm-env foo-operator]# operator-sdk version
operator-sdk version: "v0.17.1", commit: "6d108056f39fa19546cf9235fad6e84b0683114a", kubernetes version: "unknown", go version: "go1.13.10 linux/amd64"
[root@preserve-olm-env foo-operator]# ansible --version
ansible 2.9.0
```

**Comment by joelanford** [comment]
The correct version for OpenShift 4.5 is `v0.17.2` (there was an issue during the release for v0.17.1 that forced us to make a subsequent patch release).

> I test this doc with the v0.17.1 version but got the below errors. Is it a bug? Thanks!

^^ I will double check this.

**Comment by joelanford** [comment]
>>    I test this doc with the v0.17.1 version but got the below errors. Is it a bug? Thanks!

> ^^ I will double check this.


I am not able to reproduce this. With v0.17.2, I ran the following successfully:

```console
$ operator-sdk new foo-operator --repo=github.com/foo/foo-operator
INFO[0000] Creating new Go operator 'foo-operator'.
INFO[0000] Created go.mod
INFO[0000] Created tools.go
INFO[0000] Created cmd/manager/main.go
INFO[0000] Created build/Dockerfile
INFO[0000] Created build/bin/entrypoint
INFO[0000] Created build/bin/user_setup
INFO[0000] Created deploy/service_account.yaml
INFO[0000] Created deploy/role.yaml
INFO[0000] Created deploy/role_binding.yaml
INFO[0000] Created deploy/operator.yaml
INFO[0000] Created pkg/apis/apis.go
INFO[0000] Created pkg/controller/controller.go
INFO[0000] Created version/version.go
INFO[0000] Created .gitignore
INFO[0000] Validating project
go: downloading github.com/operator-framework/operator-sdk v0.17.2
INFO[0009] Project validation successful.
INFO[0009] Project creation complete.

$ operator-sdk add api --api-version=foo.com/v1 --kind=Foo
INFO[0000] Generating api version foo.com/v1 for kind Foo.
INFO[0000] Created pkg/apis/foo/group.go
INFO[0000] Created pkg/apis/foo/v1/foo_types.go
INFO[0001] Created pkg/apis/addtoscheme_foo_v1.go
INFO[0001] Created pkg/apis/foo/v1/register.go
INFO[0001] Created pkg/apis/foo/v1/doc.go
INFO[0001] Created deploy/crds/foo.com_v1_foo_cr.yaml
INFO[0001] Running deepcopy code-generation for Custom Resource group versions: [foo:[v1], ]
INFO[0005] Code-generation complete.
INFO[0005] Running CRD generator.
INFO[0006] CRD generation complete.
INFO[0006] API generation complete.

$ operator-sdk generate crds
INFO[0000] Running CRD generator.
INFO[0000] CRD generation complete.

$ ls -l deploy/crds
total 8
-rw-r--r-- 1 joe joe 1464 Jul  2 10:28 foo.com_foos_crd.yaml
-rw-r--r-- 1 joe joe   99 Jul  2 10:28 foo.com_v1_foo_cr.yaml
```

**Comment by jianzhangbjz** [comment]
@joelanford Thanks for your confirmation! Yes, it did work well for the `Go` type operator, but not for the `Ansible`/`Helm` type. Is it as expected? Or is it a bug? Thanks!  
Ansible:
```console
[root@preserve-olm-env foo-operator]# operator-sdk new --type ansible --kind Foo --api-version foo.example.com/v1alpha1 foo-operator
...

[root@preserve-olm-env foo-operator]# operator-sdk generate crds
INFO[0000] Running CRD generator.                       
FATA[0000] error generating CRDs from APIs in pkg/apis: error generating CRD manifests: error reading CRD cache dir deploy/crds: open deploy/crds: file does not exist 

[root@preserve-olm-env foo-operator]# ls deploy/crds/
foo.example.com_foos_crd.yaml  foo.example.com_v1alpha1_foo_cr.yaml
```
Helm:
```console
[root@preserve-olm-env example-inc]#  operator-sdk new --type helm --kind Foo --api-version foo.example.com/v1alpha1 foo-operator
INFO[0000] Creating new Helm operator 'foo-operator'.   
INFO[0000] Created helm-charts/foo                      
INFO[0000] Generating RBAC rules                        
WARN[0002] The RBAC rules generated in deploy/role.yaml are based on the chart's default manifest. Some rules may be missing for resources that are only enabled with custom values, and some existing rules may be overly broad. Double check the rules generated in deploy/role.yaml to ensure they meet the operator's permission requirements. 
INFO[0002] Created build/Dockerfile                     
INFO[0002] Created watches.yaml                         
INFO[0002] Created deploy/service_account.yaml          
INFO[0002] Created deploy/role.yaml                     
INFO[0002] Created deploy/role_binding.yaml             
INFO[0002] Created deploy/operator.yaml                 
INFO[0002] Created deploy/crds/foo.example.com_v1alpha1_foo_cr.yaml 
INFO[0002] Generated CustomResourceDefinition manifests. 
INFO[0002] Project creation complete.                   
[root@preserve-olm-env example-inc]# cd foo-operator/
[root@preserve-olm-env foo-operator]# ls deploy/crds/
foo.example.com_foos_crd.yaml  foo.example.com_v1alpha1_foo_cr.yaml
[root@preserve-olm-env foo-operator]# operator-sdk generate crds
INFO[0000] Running CRD generator.                       
FATA[0000] error generating CRDs from APIs in pkg/apis: error generating CRD manifests: error reading CRD cache dir deploy/crds: open deploy/crds: file does not exist 
```

Version:
```console
[root@preserve-olm-env foo-operator]# operator-sdk version
operator-sdk version: "v0.17.2", commit: "0258db0119e8e18e15d035532427c329fce1e871", kubernetes version: "unknown", go version: "go1.13.10 linux/amd64"
```

**Pull request merged**